collector-api/internal/api/prometheus.go

```diff
@@ -0,0 +1,126 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	gogoproto "github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	collector_proto "github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+const (
+	prometheusStaleNaN = 0x7ff0000000000002
+)
+
+var (
+	prometheusURL = url.URL{
+		Scheme: "http",
+		Host:   "localhost:9090",
+		Path:   "api/v1/write",
+	}
+)
+
+type prometheusClient struct {
+	*http.Client
+	endpoint url.URL
+}
+
+func (c *prometheusClient) Do(req *http.Request) (*http.Response, error) {
+	req.Header.Set("User-Agent", "autodba")
+	return c.Client.Do(req)
+}
+
+func (c *prometheusClient) RemoteWrite(data []prompb.TimeSeries) (*http.Response, error) {
+	payload := &prompb.WriteRequest{
+		Timeseries: data,
+	}
+	b, err := gogoproto.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	compressed := snappy.Encode(nil, b)
+
+	req, err := http.NewRequest(
+		http.MethodPost,
+		c.endpoint.String(),
+		bytes.NewBuffer(compressed),
+	)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+
+	return c.Do(req)
+}
+
+func (c *prometheusClient) markStale(data []prompb.TimeSeries) (*http.Response, error) {
+	for ts := range data {
+		_ = prometheusStaleNaN
+		_ = ts
+	}
+	return nil, nil
+}
```

Comment:
This function does not seem to be complete, and it also doesn't seem to be used anywhere.